xdsl/transforms/stencil_to_csl_stencil.py

```diff
@@ -171,10 +295,314 @@ def match_and_rewrite(self, op: dmp.SwapOp, rewriter: PatternRewriter, /):
             nested_rewriter.rewrite_op(new_apply_op)
 
 
+def get_op_split(
+    ops: Sequence[Operation], prefetch: BlockArgument
+) -> tuple[Sequence[Operation], Sequence[Operation]]:
+    """
+    Returns a split of ops into an `(a,b)` tuple, such that:
+
+    - `a` contains neighbour accesses plus the minumum set of instructions to reduce the accessed data to 1 thing
+    - `b` contains everything else
+
+    If no valid split can be found, return `(ops, [])`.
+
+    This function does not attempt to arithmetically re-structure the computation to obtain a good split. To do this,
+    `RestructureSymmetricReductionPattern()` may be executed first.
+    """
```

Comment:
This is a good start, although I would like to see you mention what each input argument represents (currently `prefetch` is unexplained)

Reply:
Ok